src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -321,25 +338,27 @@ private void BuildLinkTag(IDictionary<string, string> attributes, TagHelperConte
                 var attributeValue = attribute.Value;
                 if (string.Equals(attribute.Key, HrefAttributeName, StringComparison.OrdinalIgnoreCase))
                 {
-                    attributeValue = HtmlEncoder.HtmlEncode(
-                        ShouldAddFileVersion() ?
-                            _fileVersionProvider.AddFileVersionToPath(attributeValue) :
-                            attributeValue);
+                    var attributeStringValue = attributeValue as string;
+
+                    // "href" values come from bound attributes and globbing. Must always be non-null strings.
+                    Debug.Assert(attributeStringValue != null);
```

Comment:
https://github.com/aspnet/Mvc/blob/dougbu/bound.not.encoded.razor.335/src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs#L213

and if `Src` is not bound but appears in the `Attributes` dictionary, we're sunk anyhow.  that upends assumptions made in most of the MVC tag helpers.